### PR TITLE
Hanger and launcher

### PR DESCRIPTION
**Pull request by ColtonBadock**
Hanger and airplane launcher completed.

### CODE REVIEW

File: AutoBB.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.neobots2903.ftcCenterstage2023;
```
Comment (toddas90) [inline AutoBB.java:1]: Should we move the auto code into these 4 files to keep Robot cleaner?

File: robot9330.java [hunk 2]
```diff
@@ -15,10 +15,28 @@ public class Robot9330 {
...
     public boolean flip;
     public Servo airplaneLauncherRelease; //The servo that release the airplane on the launcher.
     boolean planeIsLaunched = false; //Switched to true when the plane is launched.
+    public Servo pixelTrapServoOne; //First Servo for the pixel trap.
+    public Servo pixelTrapServoTwo; //Second Servo for the pixel trap.
+    boolean pixelTrapIsDown = false;
+    public DcMotor motorHangShoulder;
+    public DcMotor motorHangArm;
+    
+    //Reverse motors for atonnymus
```
Comment (toddas90) [inline robot9330.java:28]: Should be Autonomous*

File: robot9330.java [hunk 3]
```diff
@@ -43,6 +69,53 @@ public Robot9330(OpMode opMode, boolean flip) {
...
+    
+    public void lowerHanger() {
+        motorHangShoulder.setTargetPosition(0);
+        motorHangShoulder.setMode(DcMotor.RunMode.RUN_TO_POSITION);
+        motorHangShoulder.setPower(-0.05);
+    }
+    
+    //Extends hanger extender upward.//4189 is default
+    public void extendHangerExtender() {
+        motorHangArm.setTargetPosition(4189);
```
Comment (toddas90) [inline robot9330.java:101]: Should be a variable to avoid "magic numbers" in the code.

File: robot9330.java [hunk 4]
```diff
@@ -83,8 +156,330 @@ public void pause(double seconds) {
...
+            pixelTrapServoTwo.setPosition(0);
+            pixelTrapIsDown = false;
+        }
+    }
+    
+    //Tick is what the motor encoders return.
+    //One wheel rotation = 280 ticks = 23.93 centimeters.
+    //Converts inch to ticks for driving forward and back.
+    public int toTicks_ForwardAndBack(double inch) {
+        return (int) Math.ceil(inch * 29.71);
```
Comment (toddas90) [inline robot9330.java:181]: Magic number 29.71, move to a variable.

File: robot9330.java [hunk 5]
```diff
@@ -83,8 +156,330 @@ public void pause(double seconds) {
...
+    }
+    
+    //Converts inch to ticks for driving left and right (Strafe).
+    public int toTicks_LeftAndRight(double inch) {
+        return (int) Math.ceil(inch * 30);
```
Comment (toddas90) [inline robot9330.java:186]: Magic number

**Review by toddas90 — APPROVED** [review]
Added a few small suggestions. Should be good as-is.

File: TeleOp1.java [hunk 6]
```diff
@@ -72,12 +76,45 @@ public void run() {
                         wheeling = false;
                     }
                 }.start();
-            }
+            }*/
             
-            //When "y" is pressed, launch the airplane.
-            if (gamepad1.y) {
+            //When "y" is pressed on gamepad2, launch the airplane.
+            if (gamepad2.y) {
```
Comment (alliecatstanz34) [inline TeleOp1.java:82]: Why are we switching game pads?

**Review by alliecatstanz34 — APPROVED** [review]
Looks good, just need a couple of code cleanup things

**Pull request merged**